### PR TITLE
__INTEL_COMPILER

### PR DESCRIPTION
**Pull request by ghost**
Error #396

### CODE REVIEW

**Comment by ghost** [comment]
reviewed libweb3core #defines:

```
libweb3core\libdevcore\Common.cpp: removed defines
libweb3core\libdevcore\RLP.cpp: removed defines
libweb3core\libdevcore\Common.h: ifdef __INTEL_COMPILER: disable warning 3682
libweb3core\libdevcore\vector_ref.h: ifdef __INTEL_COMPILER: disable warning 597
libweb3core\libdevcore\CommonData.h: ifdef __INTEL_COMPILER: template function toBigEndian
libweb3core\libdevcore\RLP.h: ifdef __INTEL_COMPILER: template function pushInt
```

File: libdevcore/Common.cpp [hunk 1]
```diff
@@ -94,7 +94,7 @@ string inUnits(bigint const& _b, strings const& _units)
 		auto i = *it;
 		if (i != _units.front() && b >= unit)
 		{
-			ret << (double(b / (unit / 1000)) / 1000.0) << " " << i;
+			ret << (b.convert_to<double>() / (unit.convert_to<double>() / 1000) / 1000.0) << " " << i;
```
Comment (bobsummerwill) [inline libdevcore/Common.cpp]: the 1000 should be 1000.0, shouldn't it?

File: libdevcore/Common.h [hunk 2]
```diff
@@ -33,6 +33,10 @@
 #define noexcept throw()
 #endif
 
+#ifdef __INTEL_COMPILER
+#pragma warning(disable:3682)
```
Comment (bobsummerwill) [inline libdevcore/Common.h]: Please could you add a comment to the right of the pragma with the warning message being suppressed?  Thanks!

File: libdevcore/vector_ref.h [hunk 3]
```diff
@@ -6,6 +6,10 @@
 #include <vector>
 #include <string>
 
+#ifdef __INTEL_COMPILER
+#pragma warning(disable:597)
```
Comment (bobsummerwill) [inline libdevcore/vector_ref.h]: Please could you add a comment to the right of the pragma with the warning message being suppressed?  Thanks!

File: libdevcore/CommonData.h [hunk 4]
```diff
@@ -120,6 +120,134 @@ inline void toBigEndian(T _val, Out& o_out)
 		o_out[i - 1] = (typename Out::value_type)(uint8_t)v;
 	}
 }
+#ifdef __INTEL_COMPILER
```
Comment (bobsummerwill) [inline libdevcore/CommonData.h]: What is the reason for needing these Intel-specific functions?   Where are they called from?
Just specific overrides for wide types which somehow "just work" for the other compilers?

**Comment by ghost** [comment]
within the top-level template function, any use of the convert_to function throws an error:

```
/home/libweb3core/libdevcore/CommonData.h(121): error: type name is not allowed
        o_out[i - 1] = (typename Out::value_type)(uint8_t)(v.convert_to<uint8_t>());
```

File: libdevcore/CommonData.h (hunk 4 above)
Comment (ghost) [inline libdevcore/CommonData.h]: the implicit conversion used in the template function does not compile using intel platform.
When the argument type is specified, we can call the convert_to member function.
In summary, the intel platform supports this "convert_to" member function, but does not support implicit conversion.
This member function is not available with the template parameter.

**Pull request merged**